test/e2e/tests/settings/state-logs-helpers.ts

```diff
@@ -0,0 +1,378 @@
+  expectedType?: string,
+  actualType?: string,
+): string => {
+  switch (type) {
+    case 'missing':
+      return `❌ Missing key: ${key}`;
+    case 'new':
+      return `🆕 New key found: ${key}`;
+    case 'mismatch':
+      return `🔄 Type mismatch at ${key}: expected ${expectedType}, got ${actualType}`;
+    default:
+      return `Unknown difference: ${key}`;
+  }
+};
+
+// Minimal type definition for the specific fields we validate in the State Logs Account spec
+type MinimalStateLogsJson = {
+  metamask: {
+    identities: {
+      [key: string]: {
+        address: string;
+      };
+    };
+    internalAccounts: {
+      selectedAccount: string;
+      accounts: {
+        [key: string]: {
+          address: string;
+        };
+      };
+    };
+  };
+  [key: string]: unknown;
+};
+
+const getValueType = (value: unknown): string => {
+  if (value === null) {
+    return 'null';
+  }
+  if (Array.isArray(value)) {
+    return 'array';
+  }
+  if (typeof value === 'object') {
+    return 'object';
+  }
+  return typeof value;
+};
+
+// Function to create a type-only version of an object (keys with their types)
+export const createTypeMap = (
+  obj: Record<string, unknown>,
+  path = '',
+): Record<string, string> => {
+  const typeMap: Record<string, string> = {};
+
+  for (const key in obj) {
+    if (Object.prototype.hasOwnProperty.call(obj, key)) {
+      const currentPath = path ? `${path}.${key}` : key;
+      const value = obj[key];
+      const type = getValueType(value);
+
+      typeMap[currentPath] = type;
+
+      if (type === 'object' && value !== null) {
+        const objectKeys = Object.keys(value as Record<string, unknown>);
+        if (objectKeys.length > 0) {
+          Object.assign(
+            typeMap,
+            createTypeMap(value as Record<string, unknown>, currentPath),
+          );
+        }
+        // Empty objects will still have their type recorded as 'object'
+      } else if (type === 'array' && Array.isArray(value) && value.length > 0) {
+        // For arrays, we'll check the type of the first element as a sample
+        const firstElementType = getValueType(value[0]);
+        typeMap[`${currentPath}[0]`] = firstElementType;
+
+        // If first element is an object, recursively process it
+        if (firstElementType === 'object' && value[0] !== null) {
+          Object.assign(
+            typeMap,
+            createTypeMap(
+              value[0] as Record<string, unknown>,
+              `${currentPath}[0]`,
+            ),
+          );
+        }
+      }
+    }
+  }
+
+  return typeMap;
+};
+
+// We can ignore keys for 2 reasons:
+// 1. To avoid failing for frequent state changes, which are low risk
+// 2. To mitigate flakiness for properties which appear intermittently on state, right after login in
+const getIgnoredKeys = (): string[] => [
+  'localeMessages',
+  'metamask.slides',
+  'metamask.currentBlockGasLimitByChainId',
+  'metamask.networkConfigurationsByChainId',
+];
```

Comment:
:information_source: Added a request for removing the localeMessages altogether from the state logs. Ticket is [here](https://github.com/MetaMask/metamask-extension/issues/35293). In the meantime, we can just ignore it